datasource/src/main/scala/quasar/blobstore/azure/rx.scala

```diff
@@ -29,13 +29,13 @@ import io.reactivex.observers.DisposableSingleObserver
 import io.reactivex.subscribers.DefaultSubscriber
 
 object rx {
-  final class AsyncSubscriber[A](cb: Either[Throwable, Option[A]] => Unit) extends DefaultSubscriber[A] {
+  final class AsyncSubscriber[F[_]: Sync, A](cb: Either[Throwable, Option[F[A]]] => Unit) extends DefaultSubscriber[A] {
 
     override def onStart(): Unit =
       request(1)
 
     def onNext(a: A): Unit = {
-      cb(Right(a.some))
+      cb(Right(Some(Sync[F].delay(request(1)).as(a))))
       request(1)
```

Comment:
Since we're delaying the request of the next event to when the value is handled, we no longer need to request here.

Reply:
oh, I thought I deleted that